source/exe/process_wide.cc

```diff
@@ -15,10 +16,20 @@ namespace {
 std::atomic<uint32_t>& processWideInitialized() {
   MUTABLE_CONSTRUCT_ON_FIRST_USE(std::atomic<uint32_t>);
 };
+
+// Per absl documentation it is safe for this to be a namespace-scoped global variable.
+absl::once_flag create_once;
+
 } // namespace
 
 ProcessWide::ProcessWide() {
-  if (processWideInitialized()++ == 0) {
+  // Increment the process count atomic so shutdown can be tracked.
+  processWideInitialized()++;
+
+  // Make sure that any racing calls allow only a single initializer but block all other callers
+  // until initialization is complete. This can happen in certain library use cases such as
+  // Envoy Mobile.
```

Comment:
the use-case makes sense to me but something doesn't feel quite right about the implementation.

Do we want to be able to have this sequence work?
  * Construct
  * Destruct
  * Construct
  * Destruct

I feel like absl::call_once might not allow the second construction to actually do the initialization, and the data will be left in a destructed state.

WDYT of having a construct-on-first-use absl::Mutex to protect the constructor and destructor?

Reply:
Hmm, good point. Sure, let me switch this.